finesse/gui/temp_control.py

```diff
@@ -0,0 +1,311 @@
+        self._canvas.mpl_connect("button_press_event", self._update_figure)
+
+        self._figure_num_pts = 10
+        t = [None] * self._figure_num_pts
+        hot_bb_temp = [None] * self._figure_num_pts
+        cold_bb_temp = [None] * self._figure_num_pts
+
+        colours = plt.rcParams["axes.prop_cycle"]
+        hot_colour = colours.by_key()["color"][0]
+        cold_colour = colours.by_key()["color"][1]
+
+        self._ax["hot"].plot(
+            t, hot_bb_temp, color=hot_colour, marker="x", linestyle="-"
+        )
+        self._ax["hot"].set_ylabel("HOT BB", color=hot_colour)
+
+        self._ax["cold"] = self._ax["hot"].twinx()
+        self._ax["cold"].plot(
+            t, cold_bb_temp, color=cold_colour, marker="x", linestyle="-"
+        )
+        self._ax["cold"].set_ylabel("COLD BB", color=cold_colour)
+
+        self._canvas.draw()
+
+    def _toggle_axis_visibility(self, name: str) -> None:
+        """Shows or hides BB plots."""
+        state = self._ax[name].yaxis.get_visible()
+        self._btns[name].setFlat(state)
+        self._ax[name].yaxis.set_visible(not state)
+        self._ax[name].lines[0].set_visible(not state)
+        self._canvas.draw()
+
+    def _update_figure(self, event) -> None:
+        """Updates the matplotlib figure to be contained within the panel."""
+        time = list(self._ax["hot"].lines[0].get_xdata())
+        hot_data = list(self._ax["hot"].lines[0].get_ydata())
+        cold_data = list(self._ax["cold"].lines[0].get_ydata())
+
+        time.pop(0)
+        hot_data.pop(0)
+        cold_data.pop(0)
+
+        new_time, new_hot_data, new_cold_data = get_temperature_data()
+        time.append(new_time)
+        hot_data.append(new_hot_data)
+        cold_data.append(new_cold_data)
+
+        self._ax["hot"].lines[0].set_xdata(time)
+        self._ax["hot"].lines[0].set_ydata(hot_data)
+        self._ax["cold"].lines[0].set_xdata(time)
+        self._ax["cold"].lines[0].set_ydata(cold_data)
+
+        self._ax["hot"].relim()
+        self._ax["cold"].relim()
+        self._ax["hot"].autoscale()
+        self._ax["cold"].autoscale()
+        self._ax["cold"].set_ylim(  # Confines "cold" line to lower half of plot
+            [self._ax["cold"].get_ylim()[0], 2 * self._ax["cold"].get_ylim()[1]]
+        )
+
+        xticks = self._ax["hot"].get_xticks()
+        xticklabels = [""] * len(xticks)
+        for i in range(len(xticks)):
+            t = datetime.fromtimestamp(xticks[i])
+            xticklabels[i] = t.strftime("%H:%M:%S")
+        #        self._ax["cold"].set_xticks(xticks)
+        self._ax["hot"].set_xticklabels(xticklabels)
+
+        self._canvas.draw()
+
+
+class DP9800(QGroupBox):
+    """Widgets to view the DP9800 properties."""
+
+    def __init__(self, num_channels: int) -> None:
+        """Creates the widgets to monitor DP9800.
+
+        Args:
+            num_channels (int): Number of Pt 100 channels being monitored
```

Comment:
Not a problem, but FYI you don't need to specify to type again in the docstring. MkDocs et al. are clever enough to extract the types from the function signature